### PR TITLE
Second deliver

### PR DESCRIPTION
**Pull request by livier92**

### CODE REVIEW

File: src/components/App/App.component.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <AuthProvider>
+        <BrowserRouter>
+          <Container>
+            <NavBar />
+            <Switch>
+              <Route exact path="/" component={HomePage} />
+              <Route exact path="/home/:searchText?" component={HomePage} />
+              <Route path="/login" component={LoginPage} />
+              <Route path="/video/:id" component={VideoPlayerPage} />
+              <Protected path="/favorites" component={FavoritesPage} />
```
Comment (wdonet) [inline src/components/App/App.component.js:36]: Good you defined a protected route however I could not see it working or even a way to mark a video as favourite :/

File: src/components/VideoList/VideoList.component.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import VideoContext from '../../state/videoContext';
+
+function VideoList(props) {
+  const { dispatch } = useContext(VideoContext);
+
+  const handleClick = (video) => () => {
+    dispatch({ type: 'SET_VIDEO', payload: video });
+  };
+
+  console.log(props);
```
Comment (wdonet) [inline src/components/VideoList/VideoList.component.js:15]: do not leave"console.logs" at deliverables

File: src/pages/VideoPlayer/VideoPlayer.page.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      favorites.some((favorite) => favorite.id.videoId === video.id.videoId)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  return (
+    <MainPane>
```
Comment (wdonet) [inline src/pages/VideoPlayer/VideoPlayer.page.js:45]: oh got it, thanks to this component I was able to find the button to add a fav video, however, this seems to be hidden at the bottom of the page, probably you can fix it by playing with styles

File: src/state/videoReducer.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      };
+    case 'LOAD_FROM_STORAGE': {
+      const videos = JSON.parse(localStorage.getItem('videos')) || [];
+      return {
+        ...state,
+        videos: videos,
+      };
+    }
+    case 'ADD_FAVORITE': {
+      console.log('add reducer');
```
Comment (wdonet) [inline src/state/videoReducer.js:35]: use logs only at development time

**Review by wdonet — CHANGES_REQUESTED** [review]
Just a few comments, it seems you made it 👌🏽 , however, work on CSS styling to make the videos at the list to look at the right size instead of being stretched, assure all buttons are visible using the current viewport

File: src/components/NavBar/NavBar.component.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          className=" mr-sm-2"
+        />
+        <Link to={{ pathname: `/home/${inputValue}` }} className="btn btn-primary">
+          Search
+        </Link>
+      </Form>
+      <Form inline>
+        <Nav className="mr-auto">
+          {authenticated ? (
+            <LogoutSpan className="nav-link" onClick={handleLogout} href="/">
```
Comment (jparciga) [inline src/components/NavBar/NavBar.component.js:54]: I think that the `href` attribute is not necessary here

File: src/index.css [hunk 6]
```diff
@@ -0,0 +1,11 @@
+body {
+  margin: 0;
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
```
Comment (jparciga) [inline src/index.css:3]: Please make sure to embed this font files from Google Fonts or from file declared on a `@font-face {}` statement

File: src/utils/hooks/useYoutubeAPI.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+const API_URL = 'https://youtube.googleapis.com/youtube/v3/search?';
+
+function UseYoutubeAPI(queryParam) {
+  const { state, dispatch } = useContext(VideoContext);
+
+  useEffect(() => {
+    function start() {
+      window.gapi.client
+        .init({
+          apiKey: process.env.REACT_APP_API_KEY,
```
Comment (jparciga) [inline src/utils/hooks/useYoutubeAPI.js:15]: 👌🏻

**Review by jparciga — COMMENTED** [review]
The app is working, good job! It would've been great that you have implemented tests for your components, hooks and utils. You forgot to implement related videos list on the video page. Some styles are broken on mobile.

**Pull request closed without merging**